### PR TITLE
Add Playwright tests for the story LPD-6813

### PR DESCRIPTION
**Pull request by Mikellorza**
subtask:https://liferay.atlassian.net/browse/LPD-18491

Description
----------
Added a test to test friendly url separator feature. 

LPD-18387 is already merged in brian see: https://github.com/brianchandotcom/liferay-portal/pull/147694

~~Let's wait to https://github.com/liferay-content-management/liferay-portal/pull/5166 to avoid some conflicts, since, the created test is created in the same file.~~

### CODE REVIEW

**Comment by Mikellorza** [comment]
ci:test:playwright

**Comment by Mikellorza** [comment]
Hi @NemethNorbert ,

I have just updated the PR because of a conflict with ProductMenuPage.ts, this is a very common file and for sure we will have some conflicts with it.

File: modules/test/playwright/tests/journal-web/journalArticle.spec.ts [hunk 1]
```diff
@@ -28,6 +29,75 @@ const PERMISSIONS_LOCATORS = [
...
+
+	const urlSeparator = 'content';
+
+	await friendlyUrlInstanceSettingsPage.modifySeparator(
+		'_com_liferay_configuration_admin_web_portlet_InstanceSettingsPortlet_com.liferay.journal.model.JournalArticle',
+		urlSeparator
+	);
+
+	expect(
+		await page.request.get('/' + urlSeparator + '/' + articleTitle)
```
Comment (NemethNorbert) [inline modules/test/playwright/tests/journal-web/journalArticle.spec.ts:83]: this page.request expect statement is failing on my side. Could you take a look a please?

File: modules/test/playwright/pages/layout-page-template-admin-web/PageTemplatePage.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+	async goToDisplayPageTemplates() {
+		await this.goto();
+		await this.page
+			.getByRole('link', {
+				name: 'Display Page Templates',
+			})
+			.click();
+
+		await this.page.getByText('Display Pages', {exact: true});
```
[the remark this answers is not in the document]
Comment (Mikellorza) [reply modules/test/playwright/pages/layout-page-template-admin-web/PageTemplatePage.ts:31]: You are right. I have changed it

File: modules/test/playwright/tests/journal-web/journalArticle.spec.ts [hunk 3]
```diff
@@ -28,6 +29,75 @@ const PERMISSIONS_LOCATORS = [
...
+	).toBeSuccessful();
+
+	await friendlyUrlInstanceSettingsPage.goto();
+
+	await friendlyUrlInstanceSettingsPage.resetSeparator(
+		'Web Content URL Separator'
+	);
+
+	expect(await page.request.get('/w/' + articleTitle)).toBeSuccessful();
```
[the remark this answers is not in the document]
Comment (Mikellorza) [reply modules/test/playwright/tests/journal-web/journalArticle.spec.ts:92]: @NemethNorbert ,

In addition to above, I have to know if :

1. Has the instance setting for the separator been reset correctly ?

**Comment by Mikellorza** [comment]
ci:test:playwright

**Comment by Mikellorza** [comment]
Hey @NemethNorbert ,

I have executed the test again in my local I have seen that there is an error when trying to publish a new DPT seethe [fix](https://github.com/liferay-content-management/liferay-portal/pull/5179/commits/238c6bc456fae7bcacff67728203f687f0788f8b) , maybe this also failed on your side and for that reason you could not get a successful execution, please take a look and let me know. The problem of this test is the execution, some times take more time to do a step and fails in the next, for that reason I have tried to add some wait for something  to avoid those problems.

**Comment by NemethNorbert** [comment]
> Hey @NemethNorbert ,
> 
> I have executed the test again in my local I have seen that there is an error when trying to publish a new DPT seethe [fix](https://github.com/liferay-content-management/liferay-portal/pull/5179/commits/238c6bc456fae7bcacff67728203f687f0788f8b) , maybe this also failed on your side and for that reason you could not get a successful execution, please take a look and let me know. The problem of this kind of test is the execution, some times take more time to do a step and fails in the next, for that reason I have tried to add some wait for something to avoid those problems.

Thanks great timing, I am just about to get back to reviewing this :)

**Comment by Mikellorza** [comment]
ci:test:playwright

**Comment by Mikellorza** [comment]
ci:test:playwright

**Comment by sandrodw3** [comment]
FYI: This will conflict with https://github.com/brianchandotcom/liferay-portal/pull/148203, that is already merged. If you guys have any doubt regarding it please reach me 🥰

**Comment by Mikellorza** [comment]
Okey, @sandrodw3 , thanks for notifying us  🥰

**Comment by Mikellorza** [comment]
ci:test:playwright

**Comment by sandrodw3** [comment]
@Mikellorza @NemethNorbert [test passes](https://testray.liferay.com/reports/production/logs/2024-03/test-1-32/test-portal-acceptance-pullrequest(master)/3972/playwright-js-tomcat90-mysql57-jdk8/8/0/playwright-report/index.html?authuser=0#?testId=f90f4237a88d6c2e6060-e12affa16eb91bb9ab6e) 🥳 but I'm sending a follow-up trying to remove flakiness for failing journalTemplate test: https://github.com/liferay-content-management/liferay-portal/pull/5257

**Comment by sandrodw3** [comment]
ci:close

**Pull request closed without merging**